rust/connlib/shared/src/interface/linux.rs

```diff
@@ -0,0 +1,296 @@
+                self.connection.insert(connection)
+            }
+            Some(x) => x,
+        };
+
+        let handle = &connection.handle;
+        let index = handle
+            .link()
+            .get()
+            .match_name(IFACE_NAME.to_string())
+            .execute()
+            .try_next()
+            .await?
+            .ok_or_else(|| anyhow!("No interface"))?
+            .header
+            .index;
+
+        let ips = handle
+            .address()
+            .get()
+            .set_link_index_filter(index)
+            .execute();
+
+        ips.try_for_each(|ip| handle.address().del(ip).execute())
+            .await?;
+
+        handle.link().set(index).mtu(DEFAULT_MTU).execute().await?;
+
+        let res_v4 = handle.address().add(index, ipv4.into(), 32).execute().await;
+        let res_v6 = handle
+            .address()
+            .add(index, ipv6.into(), 128)
+            .execute()
+            .await;
+
+        handle.link().set(index).up().execute().await?;
+
+        if res_v4.is_ok() {
+            if let Err(e) = make_rule(handle).v4().execute().await {
+                if !matches!(&e, NetlinkError(err) if err.raw_code() == FILE_ALREADY_EXISTS) {
+                    tracing::warn!(
+                        "Couldn't add ip rule for ipv4: {e:?}, ipv4 packets won't be routed"
+                    );
+                }
+                // TODO: Be smarter about this
+            } else {
+                tracing::debug!("Successfully created ip rule for ipv4");
+            }
+        }
+
+        if res_v6.is_ok() {
+            if let Err(e) = make_rule(handle).v6().execute().await {
+                if !matches!(&e, NetlinkError(err) if err.raw_code() == FILE_ALREADY_EXISTS) {
+                    tracing::warn!(
+                        "Couldn't add ip rule for ipv6: {e:?}, ipv6 packets won't be routed"
+                    );
+                }
+                // TODO: Be smarter about this
+            } else {
+                tracing::debug!("Successfully created ip rule for ipv6");
+            }
+        }
+
+        res_v4.or(res_v6)?;
+
+        Ok(())
+    }
+
+    pub async fn control_dns(&self, dns_config: Vec<IpAddr>) -> Result<()> {
+        match self.dns_control_method {
+            None => Ok(()),
+            Some(DnsControlMethod::EtcResolvConf) => etc_resolv_conf::configure(&dns_config).await,
+            Some(DnsControlMethod::NetworkManager) => configure_network_manager(&dns_config),
+            Some(DnsControlMethod::Systemd) => configure_systemd_resolved(&dns_config).await,
+        }
+        .context("Failed to control DNS")
+    }
+
+    #[tracing::instrument(level = "trace", skip(self))]
+    pub async fn on_update_routes(&mut self, ipv4: Vec<Cidrv4>, ipv6: Vec<Cidrv6>) -> Result<()> {
+        let new_routes: HashSet<IpNetwork> = ipv4
+            .into_iter()
+            .map(|x| Into::<Ipv4Network>::into(x).into())
```

Comment:
```suggestion
            .map(Ipv4Addr::from)
            .map(Ipv4Network::from)
```

YMMV but this seems a bit simpler.